src/CallSettings.php

```diff
@@ -191,16 +145,36 @@ public function __construct($settings = [])
         }
     }
 
-    public function getTimeoutMillis()
+    /**
+     * Creates a new instance of CallSettings that updates the settings in the existing instance
+     * with the settings specified in the $settings parameter.
+     *
+     * @param array $settings {
+     *    Optional.
+     * @type \Google\GAX\RetrySettings $retrySettings
+     *          Retry settings to use for CallSettings object.
+     * @type array $userHeaders
+     *          An array of headers to be included in the request.
+     * }
+     * @return CallSettings
+     */
+    public function with(array $settings)
     {
-        return $this->timeoutMillis;
+        $existingSettings = $this->toArray();
+        return new CallSettings($settings + $existingSettings);
```

Comment:
I'm not sure how exactly this works in PHP, but won't `$existingSettings` override items (with same key) in `$settings` (because it comes second in the expression)? Is it the case? If no, then please discard this comment. If it is, Is it the desired behavior (according to the documentation of this method it is not)?

Reply:
https://secure.php.net/manual/en/language.operators.array.php

> The + operator returns the right-hand array appended to the left-hand array; for keys that exist in both arrays, the elements from the left-hand array will be used, and the matching elements from the right-hand array will be ignored.